From 4dad6b5b6c86773d098d4dc7599078ba3d883a91 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Wed, 15 Mar 2023 20:17:39 -0400
Subject: [PATCH 02/15] [installer] MBR Updates

- Fix MBR install by using noexec=off kernel option (YAY! No more installing twice)
- Update legacy boot options to reflect recent changes on EFI side.
---
 boot/isolinux/isolinux.cfg | 51 ++++++++++++++++++++++++++++++++------
 install/scripts/1-install  | 13 +++++++---
 2 files changed, 54 insertions(+), 10 deletions(-)

diff --git a/boot/isolinux/isolinux.cfg b/boot/isolinux/isolinux.cfg
index b9311907..0a47c6e0 100644
--- a/boot/isolinux/isolinux.cfg
+++ b/boot/isolinux/isolinux.cfg
@@ -16,6 +16,43 @@ label livem
 	kernel /kernel
 	append initrd=/initrd.img CMDLINE quiet SRC= DATA=
 
+label live_intel
+	menu label PC-Mode (Intel)
+	kernel /kernel
+	append initrd=/initrd.img CMDLINE quiet SRC= DATA= HWC=drm_minigbm_celadon GRALLOC=minigbm
+
+label live_pc_intel
+	menu label PC-Mode (Intel) w/ FFMPEG
+	kernel /kernel
+	append initrd=/initrd.img CMDLINE quiet SRC= DATA= HWC=drm_minigbm_celadon GRALLOC=minigbm FFMPEG_CODEC=1 FFMPEG_PREFER_C2=1
+
+label install
+	menu label ^Installation - Install OS_TITLE to harddisk
+	kernel /kernel
+	append initrd=/initrd.img CMDLINE INSTALL=1 DEBUG= noexec=off
+
+menu separator
+
+menu begin adv_options
+menu label ^VM and other options...
+menu title VM and other options
+
+label qemu_kvm
+	menu label QEMU/KVM - Virgl - SW-FFMPEG
+	kernel /kernel
+	append initrd=/initrd.img CMDLINE quiet SRC= DATA= HWC=drm_minigbm GRALLOC=minigbm_arcvm
+
+label nomodeset
+	menu label Vbox/VMWare - No HW Acceleration
+	kernel /kernel
+	append initrd=/initrd.img CMDLINE quiet SRC= DATA= nomodeset HWACCEL=0
+
+menu separator
+
+menu begin debugging
+menu label ^Debugging options...
+menu title Debugging options
+
 label debug
 	menu label Live CD - ^Debug mode
 	kernel /kernel
@@ -41,6 +78,11 @@ label debug_minigbm
 	kernel /kernel
 	append initrd=/initrd.img CMDLINE DEBUG=2 SRC= DATA= GRALLOC=minigbm
 
+label debug_hwc_minigbm_celadon
+	menu label Live CD - Debug mode hwcomposer.drm_minigbm gralloc.minigbm
+	kernel /kernel
+	append initrd=/initrd.img CMDLINE DEBUG=2 SRC= DATA= HWC=drm_minigbm_celadon GRALLOC=minigbm
+
 label debug_hwc_minigbm
 	menu label Live CD - Debug mode hwcomposer.drm_minigbm gralloc.minigbm
 	kernel /kernel
@@ -51,11 +93,6 @@ label debug_minigbm_gbm_mesa
 	kernel /kernel
 	append initrd=/initrd.img CMDLINE DEBUG=2 SRC= DATA= HWC=drm_minigbm GRALLOC=minigbm_gbm_mesa
 
-label install
-	menu label ^Installation - Install OS_TITLE to harddisk
-	kernel /kernel
-	append initrd=/initrd.img CMDLINE INSTALL=1 DEBUG=
-
 menu separator
 
 menu begin advanced
@@ -77,12 +114,12 @@ menu separator
 label auto_install
 	menu label Auto_^Installation - Auto Install to specified harddisk
 	kernel /kernel
-	append initrd=/initrd.img CMDLINE AUTO_INSTALL=0 DEBUG=
+	append initrd=/initrd.img CMDLINE AUTO_INSTALL=0 DEBUG= noexec=off
 
 label auto_update
 	menu label Auto_^Update - Auto update OS_TITLE
 	kernel /kernel
-	append initrd=/initrd.img CMDLINE AUTO_INSTALL=update DEBUG=
+	append initrd=/initrd.img CMDLINE AUTO_INSTALL=update DEBUG= noexec=off
 
 menu separator
 
diff --git a/install/scripts/1-install b/install/scripts/1-install
index 02827ff9..6fdadf48 100644
--- a/install/scripts/1-install
+++ b/install/scripts/1-install
@@ -255,11 +255,18 @@ create_menulst()
 	echo -e "${GRUB_OPTIONS:-default=0\ntimeout=6\nsplashimage=/grub/android-x86.xpm.gz\n}root (hd0,$1)\n" > $menulst
 
 	create_entry "$OS_TITLE $VER" quiet $cmdline
+	create_entry "$OS_TITLE $VER (intel)" quiet $cmdline HWC=drm_minigbm_celadon GRALLOC=minigbm
+	create_entry "$OS_TITLE $VER (intel) w/ FFMPEG" quiet $cmdline HWC=drm_minigbm_celadon GRALLOC=minigbm FFMPEG_CODEC=1 FFMPEG_PREFER_C2=1
+	create_entry "$OS_TITLE $VER PC-Mode (intel) w/FFMPEG" quiet $cmdline PC_MODE=1 HWC=drm_minigbm_celadon GRALLOC=minigbm FFMPEG_CODEC=1 FFMPEG_PREFER_C2=1
+	create_entry "$OS_TITLE $VER PC-Mode" quiet $cmdline PC_MODE=1
+	create_entry "$OS_TITLE $VER QEMO/KVM" quiet $cmdline HWC=drm_minigbm GRALLOC=minigbm_arcvm FFMPEG_HWACCEL_DISABLE=1
+	create_entry "$OS_TITLE $VER VBOX/VMWARE" quiet $cmdline nomodeset HWACCEL=0
 	create_entry "$OS_TITLE $VER (Debug mode)" $cmdline DEBUG=2
 	create_entry "$OS_TITLE $VER (Debug mode) gralloc.gbm" $cmdline DEBUG=2 GRALLOC=gbm
-	create_entry "$OS_TITLE $VER (Debug mode) hwcomposer.drm gralloc.gbm" $cmdline DEBUG=2 HWC=drm GRALLOC=gbm
-	create_entry "$OS_TITLE $VER (Debug mode) hwcomposer.drm_minigbm gralloc.minigbm" DEBUG=2 HWC=drm_minigbm GRALLOC=minigbm
-	create_entry "$OS_TITLE $VER (Debug mode) hwcomposer.drm_minigbm gralloc.minigbm_gbm_mesa" DEBUG=2 HWC=drm_minigbm GRALLOC=minigbm_gbm_mesa
+	create_entry "$OS_TITLE $VER (Debug mode) hwc.drm gralloc.gbm" $cmdline DEBUG=2 HWC=drm GRALLOC=gbm
+	create_entry "$OS_TITLE $VER (Debug mode) hwc.drm_minigbm gralloc.minigbm" DEBUG=2 HWC=drm_minigbm GRALLOC=minigbm
+	create_entry "$OS_TITLE $VER (Debug mode) hwc.drm_minigbm_celadon gralloc.minigbm" DEBUG=2 HWC=drm_minigbm_celadon GRALLOC=minigbm
+	create_entry "$OS_TITLE $VER (Debug mode) hwc.drm_minigbm gralloc.minigbm_gbm_mesa" DEBUG=2 HWC=drm_minigbm GRALLOC=minigbm_gbm_mesa
 	create_entry "$OS_TITLE $VER (Debug nomodeset)" nomodeset $cmdline DEBUG=2
 	create_entry "$OS_TITLE $VER (Debug video=LVDS-1:d)" video=LVDS-1:d $cmdline DEBUG=2
 }
-- 
2.34.1

